crates/router/src/connector/nuvei/transformers.rs

```diff
@@ -428,43 +449,74 @@ impl<F>
             },
             _ => Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into()),
         }?;
-        Ok(Self {
-            merchant_id: merchant_id.clone(),
-            merchant_site_id: merchant_site_id.clone(),
-            client_request_id: client_request_id.clone(),
+        let request = Self::try_from(NuveiPaymentRequestData {
             amount: item.request.amount.clone().to_string(),
             currency: item.request.currency.clone().to_string(),
-            transaction_type: item
-                .request
-                .capture_method
-                .map(TransactionType::from)
-                .unwrap_or_default(),
-            time_stamp: time_stamp.clone(),
-            session_token,
-            checksum: encode_payload(vec![
-                merchant_id,
-                merchant_site_id,
-                client_request_id,
-                item.request.amount.to_string(),
-                item.request.currency.to_string(),
-                time_stamp,
-                merchant_secret,
-            ])?,
-            ..request_data
+            connector_auth_type: item.connector_auth_type.clone(),
+            client_request_id: item.attempt_id.clone(),
+            session_token: data.1,
+            capture_method: item.request.capture_method,
+            ..Default::default()
+        })?;
+        Ok(Self {
+            is_rebilling: request_data.is_rebilling,
+            user_token_id: request_data.user_token_id,
+            related_transaction_id: request_data.related_transaction_id,
+            payment_option: request_data.payment_option,
+            ..request
         })
     }
 }
+
 fn get_card_info<F>(
     item: &types::RouterData<F, types::PaymentsAuthorizeData, types::PaymentsResponseData>,
     card_details: &api_models::payments::Card,
 ) -> Result<NuveiPaymentsRequest, error_stack::Report<errors::ConnectorError>> {
     let browser_info = item.request.get_browser_info()?;
-    let related_transaction_id = if item.request.enrolled_for_3ds {
+    let related_transaction_id = if item.is_three_ds() {
         item.request.related_transaction_id.clone()
     } else {
         None
     };
-    let three_d = if item.request.enrolled_for_3ds {
+    let connector_mandate_id = &item.request.connector_mandate_id();
+    if connector_mandate_id.is_some() {
+        return Ok(NuveiPaymentsRequest {
```

Comment:
Can we avoid early returns?

Reply:
Taking this out to calling method will require more computation, is there any other to avoid this?

The logic here is, if mandate_id is present we should return the mandate details as request, otherwise return card details

Reply:
Ok, interesting, what if we use match statement, something like:
```rust
match (connector_mandate_id, setup_mandate_details) {
(Some(...), _) => { /* The above mentioned return */ },
(_, Some(...)) => { /* The code below */},
_ => { /* ... */ }
}

This will make the code look the same, but still provide a more traversable conditional branching for future reviews